alma/lib/Utils/CustomerHelper.php

```diff
@@ -0,0 +1,199 @@
+            'phone' => $this->getPhone(),
+            'country' => $addressData->getBillingCountry(),
+            'county_sublocality' => null,
+            'state_province' => $addressData->getBillingStateProvince(),
+        ];
+    }
+
+    public function getWebsiteDetails()
+    {
+        $carrierData = new CarrierData($this->context);
+        $customerOrderHelper = new CustomerOrderHelper($this->context);
+
+        return [
+            'new_customer' => $this->isNew($this->customer->id),
+            'is_guest' => (bool) $this->customer->is_guest,
+            'created' => strtotime($this->customer->date_add),
+            'current_order' => [
+                'purchase_amount' => CartData::getPurchaseAmountInCent($this->cart),
+                'payment_method' => 'alma',
+                'shipping_method' => $carrierData->getNameById($this->cart->id_carrier),
+                'items' => CartData::cartItems($this->cart),
+            ],
+            'previous_orders' => [
+                $customerOrderHelper->previous($this->customer->id),
+            ],
+        ];
+    }
+
+    /**
+     * Get birthday date
+     *
+     * @return string|null
+     */
+    private function getBirthday()
+    {
+        $birthday = $this->customer->birthday;
+        if ($birthday == '0000-00-00') {
+            return null;
+        }
+
+        return $birthday;
+    }
+
+    /**
+     * Get addresses of customer
+     *
+     * @return array getAddressesData
+     */
+    private function getAddressesData()
+    {
+        $addresses = [];
+
+        foreach ($this->getCustomerAddresses() as $address) {
+            array_push($addresses, [
+                'line1' => $address['address1'],
+                'postal_code' => $address['postcode'],
+                'city' => $address['city'],
+                'country' => Country::getIsoById((int) $address['id_country']),
+                'county_sublocality' => null,
+                'state_province' => $address['state'],
+            ]);
+        }
+
+        return $addresses;
+    }
+
+    /**
+     * Get addresses of customer Prestashop
+     *
+     * @return array getCustomerAddresses
+     */
+    private function getCustomerAddresses()
+    {
+        if (version_compare(_PS_VERSION_, '1.5.4.0', '<')) {
+            $idLang = $this->context->language->id;
+        } else {
+            $idLang = $this->customer->id_lang;
+        }
+
+        return $this->customer->getAddresses($idLang);
+    }
+
+    /**
+     * get phone between info customer and addresses customer
+     *
+     * @return string
+     */
+    private function getPhone()
+    {
+        $phone = null;
+        $addressData = new AddressData($this->cart);
+        $shippingAddress = $addressData->getShippingAddress();
+
+        if ($shippingAddress->phone) {
+            $phone = $shippingAddress->phone;
+        } elseif ($shippingAddress->phone_mobile) {
+            $phone = $shippingAddress->phone_mobile;
+        }
```

Comment:
This first part should be directly into `AddressData` class. (be careful with `new AddressData` everywhere).

```suggestion
        $addressData = new AddressData($this->cart);

        $phone = $addressData->getShippingPhone();
```